lib/discourse_api/client.rb

```diff
@@ -100,6 +100,8 @@ def connection
       @connection ||= Faraday.new connection_options do |conn|
         # Follow redirects
         conn.use FaradayMiddleware::FollowRedirects, limit: 5
+        # Allow uploading of files
+        conn.request :multipart
```

Comment:
Yes but connection is generically initialized once during each session of this client and it only uses `multipart` for files etc and not for everything.